lib/dry/validation/contract/class_interface.rb

```diff
@@ -159,7 +159,9 @@ def ensure_valid_keys(*keys)
           key_paths = key_paths(keys)
 
           invalid_keys = key_paths.map { |(key, path)|
-            unless valid_paths.any? { |vp| vp.include?(path) || vp.include?("#{path}[]") }
+            unless valid_paths.any? { |vp|
+                     vp == path || vp.include?("#{path}.") || vp.include?("#{path}[]")
+                   }
```

Comment:
I reckon we need to use a regexp here, otherwise `"footag.bar"` would have the same issue, right?

Reply:
Oh yes, you're right, I missed that case. Wouldn't `String#start_with?` also fix it while being more legible though?

Reply:
Well, it does indeed fix it, as for the more legible part, I'll let you decide ;)